applications/GeoMechanicsApplication/custom_operations/activate_model_part_operation.cpp

```diff
@@ -0,0 +1,60 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Lorenzo Gracia,
+//                   Aron Noordam,
+//                   Vahid Galavi,
+//                   Marjan Fathian
+//                   Ruben Zorrilla
+//
+
+// Project includes
+#include "containers/model.h"
+#include "includes/element.h"
+#include "includes/kratos_flags.h"
+#include "includes/kratos_parameters.h"
+#include "utilities/variable_utils.h"
+
+// Application includes
+#include "activate_model_part_operation.h"
+
+namespace Kratos
+{
+
+ActivateModelPartOperation::ActivateModelPartOperation(
+    Model& rModel,
+    const Parameters rSettings)
+    : Operation()
+    , mpModelPart(&rModel.GetModelPart(rSettings["model_part_name"].GetString()))
+{}
+
+Operation::Pointer ActivateModelPartOperation::Create(
+    Model &rModel,
+    Parameters Parameters) const
+{
+    return Kratos::make_shared<ActivateModelPartOperation>(rModel, Parameters);
+}
+
+void ActivateModelPartOperation::Execute()
+{
+    KRATOS_TRY
+
```

Comment:
In theory the mpModelPart could be a nullptr here (e.g. when there is a typo in the project parameters 'model_part_name'). It might be a good idea to throw a KRATOS_ERROR when that's the case, otherwise we'll segfault.

Reply:
Fair enough. Though a wrong model part name value would be handled by the `Model` the wrong key value (e.g. `modle_port_name` is not properly handled).

Reply:
Good point, then indeed no extra check is necessary here.